### PR TITLE
Fix test cases & Add JsonTag save functionality

### PR DESCRIPTION
**Pull request by bingcheng45**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -2,15 +2,15 @@
 layout: page title: User Guide
```
Comment (calvintanwj) [inline docs/UserGuide.md:2]: This one also same thing.
```suggestion
layout: page 
title: User Guide
```

File: src/test/java/seedu/address/logic/commands/DeleteCommandGuestTest.java [hunk 2]
```diff
@@ -1,82 +1,83 @@
...
+//
+//    /**
+//     * Updates {@code model}'s filtered list to show no one.
+//     */
+//    private void showNoPerson(Model model) {
+//        model.updateFilteredPersonList(p -> false);
+//
+//        assertTrue(model.getFilteredPersonList().isEmpty());
+//    }
+//}
```
Comment (calvintanwj) [inline src/test/java/seedu/address/logic/commands/DeleteCommandGuestTest.java:83]: This should not be commented out. It seems like you've pulled from master, and worked on it.

File: src/test/java/seedu/address/logic/commands/DeleteCommandStaffTest.java [hunk 3]
```diff
@@ -1,82 +1,82 @@
...
+//
+//    /**
+//     * Updates {@code model}'s filtered list to show no one.
+//     */
+//    private void showNoPerson(Model model) {
+//        model.updateFilteredPersonList(p -> false);
+//
+//        assertTrue(model.getFilteredPersonList().isEmpty());
+//    }
+//}
```
Comment (calvintanwj) [inline src/test/java/seedu/address/logic/commands/DeleteCommandStaffTest.java:82]: Same as above.

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 4]
```diff
@@ -80,6 +83,13 @@
             .withPassportNumber(VALID_PASSPORT_NUMBER_CARL)
             .build();
 
+    public static final Guest JEONGYEON_GUEST = new GuestBuilder()
+            .withName("Jeong Yeon")
+            .withEmail("jy@example.com")
+            .withRoomNumber("22233")
+            .withPassportNumber(PASSPORT_NUMBER_UNUSED.toString())
+            .build();
+
```
Comment (calvintanwj) [inline src/test/java/seedu/address/testutil/TypicalPersons.java:92]: Is there a need to add another person?

Comment (bingcheng45) [reply src/test/java/seedu/address/testutil/TypicalPersons.java:92]: Yes, I need to use a person to test the addition of a person, unable to reuse the others due to duplicate person

File: src/test/java/seedu/address/logic/commands/DeleteCommandGuestTest.java (hunk 2 above)
Comment (bingcheng45) [reply src/test/java/seedu/address/logic/commands/DeleteCommandGuestTest.java:83]: ah ok i see

File: docs/AboutUs.md [hunk 5]
```diff
@@ -1,6 +1,5 @@
 ---
-layout: page
-title: About Us
+layout: page title: About Us
```
[the remark this answers is not in the document]
Comment (NicolasCwy) [reply docs/AboutUs.md:2]: Note that if the changes are already found on master this isn't necessary, could cause merge conflicts which would have to be fixed later. But thanks for tidying it up!

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 6]
```diff
@@ -1,10 +1,7 @@
 package seedu.address.storage;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (NicolasCwy) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:4]: Could you revert this wildcard import to the individual imports!

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 7]
```diff
@@ -46,9 +43,12 @@ public JsonSerializableAddressBook(@JsonProperty("persons") List<Map<String, Lis
         @SuppressWarnings("unchecked")
         List<JsonAdaptedGuest> jsonGuest = (List<JsonAdaptedGuest>) persons.get(0).get("guest");
         //  this.guests.addAll(jsonGuest);
```
Comment (NicolasCwy) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:45]: Remove any commented out code you do not need in these files!

File: src/test/java/seedu/address/testutil/TypicalPersons.java (hunk 4 above)
Comment (NicolasCwy) [reply src/test/java/seedu/address/testutil/TypicalPersons.java:92]: PASSPORT_NUMBER_UNUSED is meant to guarantee that the passport number is not used by any guest. Could you create a new passport number in TypicalPassports for Jeong Yeon?

File: docs/AboutUs.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (NicolasCwy) [reply docs/AboutUs.md:2]: Thanks for these changes! Seems like someone broke the one on master

**Review by calvintanwj — APPROVED** [review]
LGTM
